app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -119,6 +119,23 @@ def ip_addresses
       []
     end
 
+    def data_disks
+      @data_disks ||= @azure_vm.storage_profile.data_disks || []
+    end
+
+    def volumes_attributes=(attrs)
+    end
+
+    def volumes
+      @volumes ||= data_disks.map do |disk|
+        OpenStruct.new(:disk => disk, :persisted? => true)
+      end
+      binding.pry
+    end
+
+    def volumes=(vols)
```

Comment:
I removed the accessor `disks` but since we set disks value here [while wrapping](https://github.com/theforeman/foreman_azure_rm/pull/71/files#diff-68325fa2c0d312fe64c53a3119247253R273) so, I had to use a setter too. It works now but, in terms of code, I wanted to know if we could keep `disks` as accessor as it is helpful to set the values since if you use `volumes` as an accessor, due to name conflicts it doesn't understand the getter method above but always considers the `[ ]`

Reply:
Note: setting volumes in AzureRmCompute.new doesn't work with this code.